Blish HUD/Controls/Container.cs

```diff
@@ -118,6 +118,60 @@ public abstract class Container : Control, IEnumerable<Control> {
             set => SetProperty(ref _autoSizePadding, value);
         }
 
+        protected Scrollbar _panelScrollbar;
+
+        protected Vector2 _scrollPadding = new Vector2(4, 0);
+        /// <summary>
+        /// Padding on the left side of the <see cref="Scrollbar"/>.
+        /// </summary>
+        public Vector2 ScrollPadding {
+            get => _scrollPadding;
+            set => SetProperty(ref _scrollPadding, value, true);
+        }
+
+        /// <summary>
+        /// Indicates whether or not there is a <see cref="Scrollbar"/>  drawn inside the <see cref="Panel"/>.
+        /// </summary>
+        public bool ScrollbarVisible {
+            //get => _panelScrollbar != null && _panelScrollbar.Drawn;
+            //get => _panelScrollbar != null;
+            get => _scrollbarVisible;
+            protected set => SetProperty(ref _scrollbarVisible, value, true);
+        }
+        protected bool _scrollbarVisible = false;
+
+        /// <summary>
+        /// Space which is taken from the <see cref="Scrollbar"/> to be drawn inside the <see cref="Panel"/>.
+        /// </summary>
+        public int ScrollbarWidth {
+            get => _panelScrollbar != null ? (_panelScrollbar.ScrollbarWidth + (int) ScrollPadding.X) : 0;
+        }
+
+        protected Point _maxSize = Point.Zero;
+        /// <summary>
+        /// Maximum Size the <see cref="Container"/> can autoresize to.
+        /// </summary>
+        public Point MaxSize {
+            get => _maxSize;
+            protected set => SetProperty(ref _maxSize, value, true);
+        }
+
+        /// <summary>
+        /// Maximum Width the <see cref="Container"/> can autoresize to.
+        /// </summary>
+        public int MaxWidth {
+            get => _maxSize.X;
+            protected set => SetProperty(ref _maxSize.X, value, true);
+        }
+
+        /// <summary>
+        /// Maximum Height the <see cref="Container"/> can autoresize to.
+        /// </summary>
+        public int MaxHeight{
+            get => _maxSize.Y;
+            protected set => SetProperty(ref _maxSize.Y, value, true);
+        }
```

Comment:
1) It currently doesn't make any sense to have all three of these.  Setting MaxWidth or MaxHeight on its own would result in a control that has either a 0 width or a 0 height since the check below only compares it against Point.Zero and doesn't check the individual dimensions.

2) You're setting the value on the structs X and Y values which I don't think is a good idea.  These would need to create a new Point like we do in the Control Size shortcuts (Width and Height properties).

3) Since sizing modes are available for both width and height individually, it seems reasonable to me to just have MaxWidth and MaxHeight.  No need for MaxSize.  They can be kept in separate `int`s anyways.  At no point do we actually use them as a point so it's just an unnecessary data type holding something we can store directly.

4) The description helps to clear things up, but it'd probably be best to apply the MaxWidth and MaxHeight to the control at all times instead of just when autoresizing.  It removes any confusion in the behavior.  That is to say that setting Size, for example, should also be limited by the MaxWidth and MaxHeight.